pointcloud/matrix_storage.go

```diff
@@ -63,13 +63,34 @@ func (ms *matrixStorage) Iterate(numBatches, myBatch int, fn func(p r3.Vector, d
 	if upperBound > ms.Size() {
 		upperBound = ms.Size()
 	}
+	// fmt.Println("numBatches: ", numBatches, "myBatch: ", myBatch, "lowerbound:", lowerBound, " upperbound:", upperBound)
 	for i := lowerBound; i < upperBound; i++ {
 		if cont := fn(ms.points[i].P, ms.points[i].D); !cont {
 			return
 		}
 	}
 }
 
+// Requires a thread safe function.
+func (ms *matrixStorage) IterateConcurrently(numBatches int, fn func(p r3.Vector, d Data) bool) {
+	if numBatches < 0 {
+		panic("should not be using this function")
+	}
```

Comment:
This is a function I wrote to iterate over the storage that backs up the kdtree so we can calculate average distance of the meanK nearest neighbors concurrently. I now realize this function is not necessary, because I could have also just called Iterate in a for loop or used the GroupWorkParallel function from parallel.go